src/classify.jl

```diff
@@ -16,3 +16,44 @@ function classify!(array, weights, mask = nothing)
     end
     array
 end
+
+"""
+    blend(arrays[, scaling])
+
+Blend arrays weighted by scaling factors.
+"""
+function blend(arrays, scaling = ones(length(arrays)))
+    ret = sum(arrays .* scaling)
+    rescale!(ret)
+end
+
+function _clusterMean(clusterArray, array)
+    clusters = Dict{Float64, Float64}()
+    clustersum = Dict{Float64, Float64}()
+    for ind in eachindex(clusterArray, array)
+        temp = clusterArray[ind]
+        if !haskey(clusters, temp)
+            clusters[temp] = clustersum[temp] = 0.0
+        end
+        clusters[temp] += 1.0
+        clustersum[temp] += array[ind]          
+    end
+    for cl in keys(clusters)
+        clustersum[cl] /= clusters[cl]
+    end
+    clustersum[NaN] = NaN
+    _rescale!(get.(Ref(clustersum), clusterArray, NaN))
+end
+
+"""
+    blendClusterArray(primary, arrays[, scaling])
+
+Blend a primary cluster NLM with other arrays in which the mean value per 
+cluster is weighted by scaling factors.
+"""
+function blendClusterArray(primary, arrays, scaling = ones(length(arrays)))
+    ret = sum(_clusterMean.(Ref(primary), arrays) .* scaling)
```

Comment:
I guess I could have just used the `blend` function too...

Reply:
I think so, the only difference is calling it with a matrix and a vector of matrices vs. a vector of matrices alone.

Reply:
so would this be preferable?
```julia
blendClusterArray(primary, arrays, scaling = ones(length(arrays))) = 
    blend([_clusterMean.(ref(primary), [primary; arrays], scaling)
```
?

Reply:
I might have a preference for the original

Reply:
my point was to maybe call *both* functions `blend`?

Reply:
ah, I see! gotcha